### PR TITLE
Fix compilation under and enable C++20

### LINKED ISSUE #12510: [MSVC][std:c++latest] terminal failed to build with /std:c++latest on MSVC
### Windows Terminal version

_No response_

### Windows build number

10.0.19041.0

### Other Software

_No response_

### Steps to reproduce

The MSVC team recently added terminal as part of RWC testing. But we found some errors when building terminal with '/std:c++latest'.

**repro steps:**
1. `git clone https://github.com/microsoft/terminal.git F:\gitP\microsoft\terminal`
2. `cd F:\gitP\microsoft\terminal`
3. `set _CL_= /std:c++latest`
4. `set _CL_=/wd4189 %_CL_%`
5. `msbuild /m /p:Platform=x64 /p:Configuration=Release /p:WindowsTargetPlatformVersion=10.0.20348.0 OpenConsole.sln /t:Restore /t:Rebuild`

### Expected Behavior

Terminal could build successfully with '/std:c++latest'.

### Actual Behavior

precomp.cpp
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt\wrl\module.h(1275,10): error C2220: the following warning is treated as an error
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt\wrl\module.h(1275,10): warning C5104: found 'L#serverName' in macro replacement list, did you mean 'L""#serverName'?
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt\wrl\module.h(1275,10): note: to simplify migration, consider the temporary use of /Wv:18 flag with the version of the compiler with which you used to build without warnings
F:\gitP\microsoft\terminal\src\inc\til\bit.h(12,1): fatal error C1021: invalid preprocessor command 'warning'
[build.log](https://github.com/microsoft/terminal/files/8086582/build.log)

### PR DESCRIPTION
**Pull request by lhecker**
This commit enables `/std:c++20` for local development under VS17.
Our CIs will continue to use VS16 and C++17 for now in order to reduce
the likelihood of regressions during the current development cycle.
It's expected that we'll migrate to VS17 soon, as this
is what conhost is already being built with anyways.

## PR Checklist
* [x] Closes #12510
* [x] I work here
* [x] Tests added/passed

## Validation Steps Performed
* Everything compiles under `/std:c++20` ✅

### CODE REVIEW

File: dep/CLI11/README.md [hunk 1]
```diff
@@ -1,5 +1,4 @@
 # CLI11
 
-Taken from [release v1.9.0](https://github.com/CLIUtils/CLI11/releases/tag/v1.9.0), source commit
-[dd0d8e4](https://github.com/CLIUtils/CLI11/commit/dd0d8e4fe729e5b1110232c7a5c9566dad884686)
-
+Taken from [release v1.9.1](https://github.com/CLIUtils/CLI11/releases/tag/v1.9.1), source commit
+[5cb3efa](https://github.com/CLIUtils/CLI11/commit/5cb3efabce007c3a0230e4cc2e27da491c646b6c)
```
Comment (DHowett) [inline dep/CLI11/README.md:4]: if there is a cgmanifest.json in the cli11 folder plz update it to match this commit

File: src/cascadia/TerminalSettingsModel/FileUtils.cpp [hunk 2]
```diff
@@ -12,7 +12,7 @@
 #include <sddl.h>
 #include <wil/token_helpers.h>
 
-static constexpr std::string_view Utf8Bom{ u8"\uFEFF" };
+static constexpr std::string_view Utf8Bom{ "\xEF\xBB\xBF", 3 };
```
[the remark this answers is not in the document]
Comment (DHowett) [reply src/cascadia/TerminalSettingsModel/FileUtils.cpp:15]: did we lose support for u8 literals? was that not a real c++ feature?

**Review by DHowett — APPROVED** [review]

File: src/cascadia/TerminalSettingsModel/FileUtils.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsModel/FileUtils.cpp:15]: Yea what's the deal here? was this just a happy accident that the compiler would translate `\uFEFF` to the utf8 `\xEF\xBB\xBF`, and now we're just being more explicit?

**Review by zadjii-msft — APPROVED** [review]

File: src/cascadia/TerminalSettingsModel/FileUtils.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (lhecker) [reply src/cascadia/TerminalSettingsModel/FileUtils.cpp:15]: This is a fallout from a breaking change in C++20 ([source](https://en.cppreference.com/w/cpp/language/string_literal)):
> u8"s-char-sequence(_optional_)"
> UTF-8 string literal. The type of a u8"..." string literal is `const char[N]` (_until C++20_) `const char8_t[N]` (_since C++20_), where `N` is the size of the string in UTF-8 code units including the null terminator.

Breaking changes in the language; still no breaking ABI change in the STL. 🥲

**Pull request merged**